Google.PowerShell.Tests/Compute/AddGceInstanceCmdletTests.cs

```diff
@@ -52,6 +53,72 @@ public void TestAcceptingCustomMemoryCustomCpu()
             Assert.AreEqual(mockedResultName, instance.Name);
         }
 
+        /// <summary>
+        /// Tests that when calling Add-GceInstance, if the request executed with
+        /// a GoogleApiException of HttpStatusCode.Conflict, a non-terminating error
+        /// is written to the pipeline.
+        /// </summary>
+        [Test]
+        public void TestErrorAddGceInstance()
+        {
+            const string psDiskVar = "attachedDisk";
+            Pipeline.Runspace.SessionStateProxy.SetVariable(psDiskVar, new AttachedDisk());
+            Mock<InstancesResource> instances = ServiceMock.Resource(s => s.Instances);
+            GoogleApiException apiException = new GoogleApiException("mock-service-name", "mock-error-message");
+            apiException.HttpStatusCode = HttpStatusCode.Conflict;
+
+            instances.SetupRequestError<InstancesResource, InstancesResource.InsertRequest, Operation>(
+                i => i.Insert(It.IsAny<Instance>(), It.IsAny<string>(), It.IsAny<string>()),
+                apiException);
+
+            Pipeline.Commands.AddScript(
+                $"Add-GceInstance -Name instance-name -Disk ${psDiskVar}");
+            Collection<PSObject> results = Pipeline.Invoke();
+
+            // An error should be thrown (if it is a terminating error,
+            // we wouldn't even reach this point).
+            Assert.AreEqual(Pipeline.Error.Count, 1);
```

Comment:
Do we want to test something about the error? Perhaps test `Pipeline.Error[0].ErrorCategory == ErrorCategory.ResourceExists`?